tokio/src/fs/file.rs

```diff
@@ -508,6 +508,35 @@ impl File {
         let std = self.std.clone();
         asyncify(move || std.set_permissions(perm)).await
     }
+
+    /// Set the maximum buffer size for the underlying [`AsyncRead`] / [`AsyncWrite`] operation.
+    ///
+    /// Although Tokio uses a sensible default value for this buffer size, this function would be
+    /// useful for changing that default depending on the situation.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use tokio::fs::File;
+    /// use tokio::io::AsyncWriteExt;
+    ///
+    /// # async fn dox() -> std::io::Result<()> {
+    /// let mut file = File::open("foo.txt").await?;
+    ///
+    /// // Set maximum buffer size to 8 MiB
+    /// file.set_max_buf_size(8 * 1024 * 1024).await;
+    ///
+    /// let mut buf = vec![1; 1024 * 1024 * 1024];
+    ///
+    /// // Write the 1 GiB buffer in chunks up to 8 MiB each.
+    /// file.write_all(&mut buf).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn set_max_buf_size(&mut self, max_buf_size: usize) {
+        let mut inner = self.inner.lock().await;
+        inner.state = State::Idle(Some(Buf::with_capacity_and_max_buf_size(0, max_buf_size)));
+    }
```

Comment:
I'm a little bit confused why exclusive access is needed for this since we're locking `inner` anyway.
Why not `&self`, lock the `inner`, await any inflight operation, and then replace the buffer?